src/cluster.py

```diff
@@ -63,11 +63,14 @@ def join_node_to_cluster(self, event: ops.framework.EventBase) -> None:
         """Join node to microceph cluster."""
         if not event.unit:
             return
+
         token = self.charm.peers.get_app_data(f"{event.unit.name}.join_token")
-        cmd = ["microceph", "cluster", "join", token]
+        if not token:
+            event.defer()
```

Comment:
We should log what's happened here, specially to debug possibly infinite loops.

Reply:
ack